dev.storyapps.starc.yml

```diff
@@ -0,0 +1,43 @@
+app-id: dev.storyapps.starc
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: starc-run
+separate-locales: false
+finish-args:
+  - --socket=x11
+  - --share=ipc
+  - --filesystem=xdg-documents:rw
+  # Needs sound to play optional typewriter sounds
+  - --socket=pulseaudio
+  - --share=network
+modules:
+  - name: starc
+    buildsystem: simple
+    build-commands:
+      - chmod +x starc-setup.AppImage
+      - ./starc-setup.AppImage --appimage-extract
+      - rm starc-setup.AppImage
+      - cp squashfs-root/starc.desktop                                      ${FLATPAK_ID}.desktop
+      - cp squashfs-root/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png ${FLATPAK_ID}.png
+      - cp squashfs-root/share/metainfo/${FLATPAK_ID}.metainfo.xml          ${FLATPAK_ID}.metainfo.xml
+      - desktop-file-edit --set-icon ${FLATPAK_ID} --set-key Exec --set-value 'starc-run %u' ${FLATPAK_ID}.desktop
+      - install -Dm644 ${FLATPAK_ID}.desktop      -t ${FLATPAK_DEST}/share/applications
+      - install -Dm644 ${FLATPAK_ID}.png          -t ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps
+      - install -Dm644 ${FLATPAK_ID}.metainfo.xml -t ${FLATPAK_DEST}/share/metainfo
+      - install -Dm755 starc-run                  -t ${FLATPAK_DEST}/bin
+      - mv squashfs-root ${FLATPAK_DEST}/bin/starc
+    sources:
+      - type: script
+        dest-filename: starc-run
+        commands:
+          - /app/bin/starc/starc $@
+      - type: file
+        filename: starc-setup.AppImage
+        url: https://github.com/story-apps/starc/releases/download/v0.3.1/starc-setup.AppImage
```

Comment:
once again, why not building from source?

Reply:
@hfiguiere Many of the gitmodules needed for building this from source aren't actually public:

https://github.com/story-apps/starc/blob/master/.gitmodules

Reply:
> why not building from source?

Because some part of the app is not open source.

Reply:
so the license is a lie?

Reply:
that's even more disappointing.

Reply:
It's dual licensed. Open source part is licensed under GPLv3 and closed source part is licensed under EULA-like license.

Reply:
it can't be GPL3 licensed of the app can't be rebuild. This is an impossibility.

Reply:
It can be built, but all of the commercial modules will be missed, but we want to include them.

Reply:
Just to be clear, there is proprietary code bundled inside of a GPL3 application?

I believe that is simply not allowed by the GPL3.

Reply:
(You can distribute commercial modules *outside* of a GPL3 project but when they are shipped together it becomes a "combined work")

Reply:
One way I think this could be packaged is building the main source in one package and having the proprietary modules be a separate extension point that the user can optionally install.

Reply:
> Just to be clear, there is proprietary code bundled inside of a GPL3 application?

No, basically, it’s a commercial app. But at some point we decided to open core part, separate it and publish then.